packages/graphql-auth-transformer/src/__tests__/AmplifyAdminAuth.test.ts

```diff
@@ -0,0 +1,128 @@
+import { GraphQLTransform } from 'graphql-transformer-core';
+import { ResourceConstants } from 'graphql-transformer-common';
+import { DynamoDBModelTransformer } from 'graphql-dynamodb-transformer';
+import { ModelAuthTransformer } from '../ModelAuthTransformer';
+
+test('Test simple model with public auth rule and amplify admin app is present', () => {
+  const validSchema = `
+    type Post @model @auth(rules: [{allow: public}]) {
+        id: ID!
+        title: String!
+        createdAt: String
+        updatedAt: String
+    }
+    `;
+  const transformer = new GraphQLTransform({
+    transformers: [
+      new DynamoDBModelTransformer(),
+      new ModelAuthTransformer({
+        authConfig: {
+          defaultAuthentication: {
+            authenticationType: 'API_KEY',
+          },
+          additionalAuthenticationProviders: [
+            {
+              authenticationType: 'AWS_IAM',
+            },
+          ],
+        },
+        addAwsIamAuthInOutputSchema: true,
+      }),
+    ],
+  });
+  const out = transformer.transform(validSchema);
+  expect(out).toBeDefined();
+  expect(out.schema).toContain('Post @aws_iam @aws_api_key');
```

Comment:
You should parse the generated document and check for directive instead of doing `toContained` similar to what we have in some other places
https://github.com/aws-amplify/amplify-cli/blob/ae304e7e2576ac6fe742cd16e9f638a425c45544/packages/graphql-auth-transformer/src/__tests__/MultiAuth.test.ts#L375-L384